specification/machinelearning/Azure.AI.ChatProtocol/operations.tsp

```diff
@@ -0,0 +1,35 @@
+import "@typespec/http";
+
+import "@azure-tools/typespec-azure-core";
+
+import "./models/blocking.tsp";
+import "./models/streaming.tsp";
+
+namespace Azure.AI.ChatProtocol;
+
+using TypeSpec.Http;
+using Azure.Core.Foundations;
+
+@route("/chat")
+interface Chat<StateModel, ContextModel> {
+  /* Using the Foundations.Operation template so the API version parameter is added */
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Defining the operation in an Azure independent fashion"
+  @doc("Creates a new streaming chat completion.")
+  @sharedRoute
+  createStreaming is Operation<
```

Comment:
I think we do not recommend directly using `Azure.Core.Foundations.Operation`?

One should use `RpcOperation` or other more appropriate ones like `ResourceCreateOrReplace`?
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-core/lib/operations.tsp#L26

Reply:
We probably can change these to use RpcOperation. As for ResourceAction, we can't really use it here as there is no resource in this API (in openai we used the deployment as the resource). Tomorrow I'll try updating these to RpcOperation

Reply:
Tried replacing this with an RpcOperation but it seems that it is causing some issues with the openapi generation. Will probably need to rollback for now.

Reply:
I see. This "x-ms-paths" likely be a bug in typespec-autorest in case of `@sharedRoute` ("x-ms-paths" should only happen when there is same path in "paths").

Sure, if it not work for now, you had to use `Operation` then.